### PR TITLE
feat(Pagination): add disabled flag for whole component

### PR DESCRIPTION
**Pull request by kmcfaul**
**What**: Adds 'isDisabled' flag to disable the whole pagination component.

**See issue**: #2315

### CODE REVIEW

**Comment by kmcfaul** [comment]
> The options menu toggle isn't aligning correctly

Am I missing a class?

File: packages/patternfly-4/react-core/src/components/Pagination/OptionsToggle.tsx [hunk 1]
```diff
@@ -41,16 +43,17 @@ export const OptionsToggle: React.FunctionComponent<OptionsToggleProps> = ({
   showToggle = true,
   onToggle = (_isOpen: boolean) => undefined as any,
   isOpen = false,
+  isDisabled = false,
   parentRef = null,
   toggleTemplate: ToggleTemplate = '',
 }:OptionsToggleProps ) => {
   return (
-    <div className={css(styles.optionsMenuToggle, getModifier(styles, 'plain'), getModifier(styles, 'text'))} >
+    <div className={css(styles.optionsMenuToggle, isDisabled && getModifier(styles, 'disabled'), getModifier(styles, 'plain'), getModifier(styles, 'text'))} >
```
Comment (jschuler) [inline packages/patternfly-4/react-core/src/components/Pagination/OptionsToggle.tsx]: nit: it generally makes more sense to use getModifier if a variable prop can be matched to the style object. In this case i would change these to `styles.disabled`, `styles.plain`, etc..

File: packages/patternfly-4/react-core/src/components/Pagination/Pagination.md [hunk 2]
```diff
@@ -87,6 +87,48 @@ class PaginationBottom extends React.Component {
 }
 ```
 
+## Pagination disabled
+```js
+import React from 'react';
+import { Pagination, PaginationVariant } from '@patternfly/react-core';
+
+class PaginationTop extends React.Component {
```
Comment (jschuler) [inline packages/patternfly-4/react-core/src/components/Pagination/Pagination.md]: nit: class name should be `PaginationDisabled`. Doesn't affect the rendering but shows up in the sample code

File: packages/patternfly-4/react-core/src/components/Pagination/OptionsToggle.tsx (hunk 1 above)
Comment (kmcfaul) [reply packages/patternfly-4/react-core/src/components/Pagination/OptionsToggle.tsx]: Ah I thought that was just how we retrieved modifier styles. I can change it to `styles.modifiers.disabled` and so on.

**Pull request merged**